app/gui2/src/components/TextFormattingSelector.vue

```diff
@@ -0,0 +1,132 @@
+<script setup lang="ts">
+import icons from '@/assets/icons.svg'
+import DropdownMenu from '@/components/DropdownMenu.vue'
+import SvgButton from '@/components/SvgButton.vue'
+import { ref } from 'vue'
+import { TextFormatOptions } from './visualizations/TableVisualization.vue'
+
+const emit = defineEmits<{
+  changeFormat: [formatValue: TextFormatOptions]
+}>()
+
+const textFormatterSelected = ref(TextFormatOptions.Partial)
+const setTextFormatterSelected = (option: TextFormatOptions) => {
+  textFormatterSelected.value = option
+  emit('changeFormat', textFormatterSelected.value)
+}
+
+const iconPath = `${icons}#paragraph`
+
+const open = ref(false)
+
+const toggleMenu = () => {
+  open.value = !open.value
+}
+</script>
+
+<template>
+  <DropdownMenu v-model:open="open" class="TextFormattingSelector">
+    <template #button
+      ><SvgButton name="paragraph" title="Text Display Options" @click.stop="toggleMenu()" />
+    </template>
+
+    <template #entries>
+      <button
+        :class="`${textFormatterSelected === TextFormatOptions.On && 'selected'}`"
+        :onclick="() => setTextFormatterSelected(TextFormatOptions.On)"
+        title="Text displayed in monaspace font and all whitespace characters displayed as a symbol"
+      >
+        <svg viewBox="0 0 16 16" width="16" height="16" style="stroke: black; fill: #000000">
+          <use :href="`${iconPath}`" />
+        </svg>
```

Comment:
Why not `SvgIcon`?

Reply:
I didn't use svg icon because it was decided one would be a grey shade of the svg and so thought this was more suited,